core/src/main/java/com/novoda/noplayer/internal/exoplayer/mediasource/MediaSourceFactory.java

```diff
@@ -44,16 +49,30 @@ public MediaSource create(Options options,
                               MediaSourceEventListener mediaSourceEventListener,
                               DefaultBandwidthMeter bandwidthMeter) {
         DefaultDataSourceFactory defaultDataSourceFactory = createDataSourceFactory(bandwidthMeter);
+
+        MediaSource contentMediaSource;
```

Comment:
maybe you can just do `MediaSource contentMediaSource = getContent...(...); so that the details on how to get this is encapsulated in a method

Reply:
https://github.com/novoda/no-player/pull/204/commits/b0b8a34c12adc4690da4dd970733c827fdf14299